app/src/RequestMoneyValidator/Dto/RequestDto.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\src\RequestMoneyValidator\Dto;
+
+
+use App\Enum\Currency;
+
+readonly final class RequestDto
+{
+    private function __construct(
+        public float $amount,
+        public Currency $currency
+    ) {}
+
+    public static function fromRequest(array $data): self
+    {
+        return new self(
+            amount: round($data['amount'], 2),
+            currency: Currency::get($data['currency']),
+        );
+    }
+
+    public function convertAmountToCents(): int
+    {
+        return $this->amount * 100;
```

Comment:
1. What if the hyperinflation takes in place? Or the currency contains not 100 subitems (there are examples with 1000 and even 5).
2. Also, touching the float (multiply then divide) is not safe – it can lead to unpredictable side effects
3. 100 - is a "magic number" here